arangod/Utils/DatabaseGuard.cpp

```diff
@@ -24,41 +24,35 @@
 #include "DatabaseGuard.h"
 #include "Basics/Exceptions.h"
 #include "RestServer/DatabaseFeature.h"
+#include "VocBase/vocbase.h"
 
-namespace {
-
-template<typename T>
-TRI_vocbase_t& vocbase(arangodb::DatabaseFeature& feature, T& id) {
-  auto* vocbase = feature.useDatabase(id);
+namespace arangodb {
 
-  if (!vocbase) {
-    THROW_ARANGO_EXCEPTION(TRI_ERROR_ARANGO_DATABASE_NOT_FOUND);
+void VocbaseReleaser::operator()(TRI_vocbase_t* vocbase) const noexcept {
+  if (vocbase) {
```

Comment:
Btw unique_ptr::deleter can be called only explicit or if stored pointer not null